test/hotspot/jtreg/runtime/cds/appcds/BootClassPathMismatch.java

```diff
@@ -235,6 +236,22 @@ public void testBootClassPathMismatchTwoJars() throws Exception {
             .assertAbnormalExit(mismatchMessage);
     }
 
+    /* Arcchive contains Hello class with only hello.jar in bootclasspath at dump time.
+     *
+     * No error - bootclasspath can be appended during runtime if no -cp is specified.
+     */
+    public void testBootClassPathAppend() throws Exception {
```

Comment:
A refactoring should not be introducing new test cases. Did you refactor and enhance?

Reply:
This is a missing testcase and works the same before and after refactoring.

Reply:
I needed to exclude the new test from running in the dynamic archive mode because there's no -cp specified during dumping. Otherwise, the test fails with`java.lang.RuntimeException: test.dynamic.dump is not supported with an empty classpath while the classlist is not empty`.